cmd/soroban-cli/src/commands/contract/invoke.rs

```diff
@@ -561,3 +555,29 @@ Each arg has a corresponding --<arg_name>-file-path which is a path to a file co
 Note: The only types which aren't JSON are Bytes and BytesN, which are raw bytes"#
     )
 }
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, ValueEnum, Default)]
+pub enum Send {
+    /// Only send transaction if there are ledger writes, otherwise return simulation result
+    #[default]
+    IfWrite,
+    /// Do not send transaction, return simulation result
+    No,
+    /// Always send transaction
+    Yes,
+}
+
+impl Send {
+    pub fn should_send(self, sim_res: &SimulateTransactionResponse) -> Result<bool, Error> {
+        Ok(match self {
+            Send::IfWrite => !sim_res
+                .transaction_data()?
+                .resources
+                .footprint
+                .read_write
+                .is_empty(),
```

Comment:
Needs to also take into account if the simulation shows that events will be published. Published events are a write.

Reply:
Ok thanks!